airflow/utils/db.py

```diff
@@ -626,27 +629,88 @@ def check_migrations(timeout):
     :param timeout: Timeout for the migration in seconds
     :return: None
     """
-    from alembic.runtime.migration import MigrationContext
+    ticker = 0
+    while True:
+        source_heads = get_source_heads()
+        db_heads = get_db_heads()
+        if source_heads == db_heads:
+            break
+        if ticker >= timeout:
+            raise TimeoutError(
+                f"There are still unapplied migrations after {ticker} seconds. "
+                f"Migration Head(s) in DB: {db_heads} | Migration Head(s) in Source Code: {source_heads}"
+            )
+        ticker += 1
+        time.sleep(1)
+        log.info('Waiting for migrations... %s second(s)', ticker)
```

Comment:
Maybe?

```suggestion
    for ticker in range(timeout):
        source_heads = get_source_heads()
        db_heads = get_db_heads()
        if source_heads == db_heads:
            return
        time.sleep(1)
        log.info('Waiting for migrations... %s second(s)', ticker)
    raise TimeoutError(
        f"There are still unapplied migrations after {timeout} seconds. "
        f"Migration Head(s) in DB: {db_heads} | Migration Head(s) in Source Code: {source_heads}"
    )
```

(This emits one extra *Waiting for migrations* line; we can eliminate that with an additional `if` check but I don't feel it's worthwhile.)

Reply:
`never use while loops` :) ?